src/services/slam/client.test.ts

```diff
@@ -0,0 +1,154 @@
+// @vitest-environment happy-dom
+
+import { afterEach, beforeEach, expect, test, vi } from 'vitest';
+import { SLAMServiceClient } from '../../gen/service/slam/v1/slam_pb_service';
+import { type ResponseStream } from '../../gen/robot/v1/robot_pb_service';
+import { RobotClient } from '../../robot';
+import { EventDispatcher } from '../../events';
+import {
+  GetInternalStateResponse,
+  GetPointCloudMapResponse,
+} from '../../gen/service/slam/v1/slam_pb';
+import { SlamClient } from './client';
+
+let slam: SlamClient;
+
+export class TestResponseStream<T> extends EventDispatcher {
+  private stream: ResponseStream<any>;
+
+  constructor(stream: ResponseStream<any>) {
+    super();
+    this.stream = stream;
+  }
+
+  override on(
+    type: string,
+    handler: (message: any) => void
+  ): ResponseStream<T> {
+    super.on(type, handler);
+    return this;
+  }
+
+  cancel(): void {
+    this.listeners = {};
+    this.stream.cancel();
+  }
+}
+
+let pcdStream: ResponseStream<GetPointCloudMapResponse>;
+let testPcdStream: TestResponseStream<GetPointCloudMapResponse> | undefined;
+let internalStream: ResponseStream<GetInternalStateResponse>;
+let testInternalStream:
+  | TestResponseStream<GetInternalStateResponse>
+  | undefined;
+
+beforeEach(() => {
+  testPcdStream = new TestResponseStream(pcdStream);
+  testInternalStream = new TestResponseStream(internalStream);
+  RobotClient.prototype.createServiceClient = vi
+    .fn()
+    .mockImplementation(() => new SLAMServiceClient('myslam'));
+
+  SLAMServiceClient.prototype.getPointCloudMap = vi
+    .fn()
+    .mockImplementation(() => testPcdStream);
+  SLAMServiceClient.prototype.getInternalState = vi
+    .fn()
+    .mockImplementation(() => testInternalStream);
+
+  slam = new SlamClient(new RobotClient('host'), 'test-slam');
+});
```

Comment:
[opt] it might be helpful to make two [describe blocks](https://vitest.dev/api/#describe) here, one for `getPointCloudMap` and one for `getInternalStream`. totally find to leave as is if it's annoying to re-organize.